Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Smoothing/ceres_support.h

```diff
@@ -36,15 +36,21 @@
 
 
 // If a user is using minilog instead of glog, the verbosity cannot be controlled via ceres options...
+// Since we really don't want to hear about it, override that value with '0'
 #ifdef MAX_LOG_LEVEL
-#  define MAX_LOG_LEVEL_WAS_PREVIOUSLY_DEFINED
-#else
-  #define MAX_LOG_LEVEL 0
+#  define MAX_LOG_LEVEL_WAS_DEFINED
+#  pragma push_macro("MAX_LOG_LEVEL")
```

Comment:
You need to protect that with `#if defined(BOOST_MSVC)`

Reply:
minilog, which causes Ceres to set that annoying macro, is multi-platform as far I see so I still need to set it up. These pragma seemed to be supported by gcc and clang, so I didn't protect this code with MSVC checks. Could you please explain what I missed?

I don't really know how else to temporarily store a macro value. Maybe it isn't worth the effort and I should just `#undef #def` and be done with it...

Reply:
You are right. I have verified: [gcc-4.4 and clang-3.0 know `push_macro`](
https://godbolt.org/z/sm8c2U). That is enough for CGAL-5.0. Thanks for you correcting me.